src/model/http-request.js

```diff
@@ -0,0 +1,36 @@
+const { Validatable } = require('../mixins/validatable-http-message');
+const clone = require('clone');
+
+class HttpRequest extends Validatable {
+  constructor(resources = {}) {
+    super();
+
+    HttpRequest.resourceKeys.forEach((resourceKey) => {
```

Comment:
Unfortunately, that doesn't work. 

I still assign `validatableComponents` like this:

```js
Validatable.validatableComponents = ['headers', 'body', 'statusCode'];
```

I can't assign it to `this`, because `gavel-spec` asserts class properties, and there is no expected `validatableComponents` property. So when `HttpRequest` extends `Validatable`, it can't reference to `this.constructor.validatableComponents`.

Reply:
Oh, I think I made a mistake by removing the `HttpRequest.resourceKeys`. Reverted, references through `this.constructor.resourceKeys` works. Thanks!